### ISSUE TITLE
Crash with Logging Out

### ISSUE THREAD

**Issue by maxmaxandr** [GetStream/stream-chat-android#467]
SDK 4.2.3
Log in as a user with id = 9
Logout and launch 
```
       chatDomain.disconnect()
       client.disconnect()
```

Log in as a user with id = 20
When running this code 
```
        val chatDomain = ChatDomain.Builder(
            context.applicationContext,
            chatClient,
            user
        )
            .offlineEnabled()
            .userPresenceEnabled()
            .build()

```
Get a crash

```
`
E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-3
    Process: com.witley, PID: 31129
    java.util.InputMismatchException: received connect event for user with id 9 while chat domain is configured for user with id 20. create a new chatdomain when connecting a different user.
        at io.getstream.chat.android.livedata.ChatDomainImpl.updateCurrentUser$library_release(ChatDomainImpl.kt:185)
        at io.getstream.chat.android.livedata.EventHandlerImpl.updateOfflineStorageFromEvents$library_release(EventHandlerImpl.kt:179)
        at io.getstream.chat.android.livedata.EventHandlerImpl$updateOfflineStorageFromEvents$1.invokeSuspend(Unknown Source:12)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
`
```
And more logs related to it

```
 I/Chat:Events: onOpen
 I/Chat:Events: onMessage: {"type":"health.check","connection_id":"47e5c45c-9c8d-4373-97bd-8c0f618f8bcf","cid":"*","me":{"id":"9","role":"user","roles":[],"created_at":"2020-05-11T05:06:36.278544Z","updated_at":"2020-05-11T11:19:23.926147Z","last_active":"2020-05-11T11:19:23.929167129Z","banned":false,"online":true,"invisible":false,"devices":[],"mutes":[],"channel_mutes":[],"unread_count":5,"total_unread_count":5,"unread_channels":2,"name":"m1 m1","image":""},"created_at":"2020-05-11T11:19:23.934337477Z"}

```

**Comment by tschellenbach** [comment]
How are you switching the user connection on the Client object? This error happens when chatDomain and client have a different user (so it's good that you're getting this error).

**Comment by maxmaxandr** [comment]
> How are you switching the user connection on the Client object? This error happens when chatDomain and client have a different user (so it's good that you're getting this error).

There is no special. User is the same for both instances
```
        val chatClient = Chat.getInstance().client

        val user = prefs.account.toSteamUser()
        chatClient.setUser(user, prefs.streamToken?.token ?: "", object : InitConnectionListener() {
            override fun onError(error: ChatError) {
                super.onError(error)
                Timber.e(error)
            }

            override fun onSuccess(data: ConnectionData) {
                super.onSuccess(data)
                systemNotifier.send(
                    UserConnected(
                        data.user.totalUnreadCount,
                        data.user.unreadChannels
                    )
                )
            }
        })

        val chatClient = Chat.getInstance().client

        val user = prefs.account.toSteamUser()
        val chatDomain = ChatDomain.Builder(
            context.applicationContext,
            chatClient,
            user
        )
           .userPresenceEnabled()
            .build()


```

**Comment by tschellenbach** [comment]
@elevenetc does the LLC perhaps cache the old connection event? IE login with user1 -> store connection event. Disconnect and after that login with user2. Connection event for user 1 is replayed? (Just a theory, didn't check)

**Comment by maxmaxandr** [comment]
And more info 

```
**//There setup user for ChatClient id=9**
V/ChatClientProvider: ChatClient init: User(id=9, role=, invisible=false, banned=false, devices=[], online=false, createdAt=null, updatedAt=null, lastActive=null, totalUnreadCount=0, unreadChannels=0, unreadCount=0, mutes=[], extraData={name=m m})

 I/Chat:SocketService: connect
I/Chat:SocketService: setupSocket
I/Chat:SocketService: updateState: Connecting

There looks like right userId
I/Chat:SocketService: httpClient.newWebSocket: wss://chat-us-east-1.stream-io-api.com/connect?json=%7B%22user_details%22%3A%7B%22role%22%3A%22%22%2C%22devices%22%3A%5B%5D%2C%22name%22%3A%22m+m%22%2C%22invisible%22%3Afalse%2C%22banned%22%3Afalse%2C%22id%22%3A%229%22%7D%2C%22user_id%22%3A%22**9**%22%2C%22server_determines_connection_id%22%3Atrue%2C%22X-STREAM-CLIENT%22%3A%221.4.4-release%22%7D&api_key=twj335e5wcah&authorization=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiMTkifQ.h_Ahw1pVu2aqFSp4G6QXG6LpbN8dZWZn0gkncAYVZ1A&stream-auth-type=jwt

**//There init ChatDomain. UserId still right id=9** 
 V/ChatDomainProvider: ChatDomain init: User(id=9, role=, invisible=false, banned=false, devices=[], online=false, createdAt=null, updatedAt=null, lastActive=null, totalUnreadCount=0, unreadChannels=0, unreadCount=0, mutes=[], extraData={name=m m})
 I/Chat:Domain: Initializing ChatDomain with version 1.4.4-release

**But there we have a wrong userId from previous user. **
 I/Chat:Notifications: FirebaseInstanceId returned token successfully
I/Chat:Notifications: setFirebaseToken: f5G2WmYeTO6_etyHwClyqR:APA91bFDdCa42RLemQgPB94KhMkltOUDpggpcdu9rv7MyrlXL-S05xJw_7kF1RHk_A3D0B0mzak0pqAvcNOwpxhj6r7sGRJqmoTz19Bx6eqN6LUtbqj6SbFRKWADGNwbiueKsJw6Qbun
I/Chat:Http: --> POST https://chat-us-east-1.stream-io-api.com/devices?api_key=twj335e5wcah&**user_id=19**&client_id=c4ece60e-5ee5-45ee-a078-1b0469112800

I/Chat:Http: X-STREAM-CLIENT: 1.4.4-release


So a bit later I will get crash

 java.util.InputMismatchException: received connect event for user with id 19 while chat domain is configured for user with id 9. create a new chatdomain when connecting a different user.
        at io.getstream.chat.android.livedata.ChatDomainImpl.updateCurrentUser$library_release(ChatDomainImpl.kt:185)
        at io.getstream.chat.android.livedata.EventHandlerImpl.updateOfflineStorageFromEvents$library_release(EventHandlerImpl.kt:179)
        at io.getstream.chat.android.livedata.EventHandlerImpl$updateOfflineStorageFromEvents$1.invokeSuspend(Unknown Source:12)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
2020-05-12 10:43:55.187 22554-22600/com.witley W/System: A resource failed to call close. 
```

**Comment by tschellenbach** [comment]
related: https://github.com/GetStream/stream-chat-android-client/pull/34

**Comment by JcMinarro** [comment]
Hello @maxmaxandr 
Could you check if it is fixed by #563 
It is already part of the version [4.2.11-beta-6](https://github.com/GetStream/stream-chat-android/releases/tag/4.2.11-beta-6), not a final release yet, we will update #554 when the final version is ready
